README.md

```diff
@@ -1,234 +1,464 @@
+- `ipv6.enabled = true`
+- `ipv6.dual_stack = true`
+- Both IPv4 and IPv6 are active within the cluster.
 
-If you do need to undo the k8s install on the VMs `./uninstall_k8s.sh` should work, but creating brand-new VMs the best way to ensure a clean slate.
+**Note**: IPv6-only clusters are not supported due to complexity and external dependencies (e.g., GitHub Container Registry lacks IPv6).
 
-## Dual Stack Networking
-The VLAN and VMs created by Tofu can have IPv6 enabled both on the host level and inside the cluster for dual-stack networking. There are three configurations for IPv6 and dual-stack networking...
-1. `ipv6.enabled = false` will disable IPv6 on the host and VLAN. Of course, the cluster will not be dual-stack enabled in this case.
-2. `ipv6.enabled = true`, but `ipv6.dual_stack = false` will enable IPv6 on the host and VLAN, but the cluster will only have IPv4 addresses. This is helpful so the hosts can resolve ipv6 addresses, but don't need dual stack services.
-3. `ipv6.enabled = true`, and `ipv6.dual_stack = true` will enable IPv6 on the host and VLAN, and the cluster will have both IPv4 and IPv6 addresses. This is the most complex configuration.
+**Tip**: The HA kube-vip API server can utilize an IPv6 address without enabling dual-stack.
 
-Currently, there is no option to have an IPv6-only cluster. This is a complex use case that complicates the setup for various reasons. For example, github's container registry doesn't have an IPv6 address.
+### Custom Worker Types
 
-*Note: The HA kube-vip apiserver address can be IPv6 without enabling dual-stack.*
+Define custom worker classes in `clusters.tf` to meet specific workload requirements:
 
-## Custom worker types
+- **GPU Workers**:
 
-You can add more custom worker types under `node_classes` in `clusters.tf`. This can be done to have k8s nodes with differing CPU, memory, disks, ip ranges, labels, taints, and devices.
+  - **Example**: Already implemented in `clusters.tf`
+  - **Use Case**: AI and machine learning workloads.
 
-Ideas for practical worker classes:
-* `gpu` class that has a GPU device for running AI workloads (this is already implemented in `clusters.tf`)
-* `storage` class with extra disks and a taint so only your storage system (i.e. Rook) runs on it
-* `database` class with increased memory
-* `fedramp` class with a taint so only government containers are run on that machine
-* `backup` class with reduced cpu and memory, a taint, and expanded disks, for only storing backups
+- **Storage Workers**:
+
+  - **Configuration**: Extra disks, taints for storage systems like Rook.
+
+- **Database Workers**:
+
+  - **Configuration**: Increased memory for database operations.
+
+- **FedRAMP Workers**:
+
+  - **Configuration**: Taints to restrict workloads to government containers.
+
+- **Backup Workers**:
+  - **Configuration**: Reduced CPU and memory, expanded disks, taints for backup storage.
+
+**Example Entry in `clusters.tf`**:
+
+```hcl
+node_classes = {
```

Comment:
Terraform requires that all the fields are set. Maybe it's best to leave this out until my next PR. I'll create default values so that a node class can be shortened like how you have it here.

Reply:
Are you suggesting this example is no longer needed after your next PR?

Reply:
It'll be needed after the next pr, but not before. It's currently a lot more complex than you make it out to be here because you would have to also edit the clusters type definition.

I'll put this section in the readme of the PR that simplifies adding node classes. I should be done with it in a few days.